### PR TITLE
refactor: Update Payment Pointer web monetization webhook 

### LINKED ISSUE #1650: Payment pointer liquidity event: update `received`
## Context

All webhook events include liquidity in the `AmountJSON` type, i.e.
```
{
  value: string
  assetCode: string
  assetScale: number
}
```
However, the payment pointer webhook event only returns `received: string`.  See [here](https://github.com/interledger/rafiki/blob/f1a396c1d4a300d220f83267e0846daaa4c6afe4/packages/backend/src/open_payments/payment_pointer/model.ts#L113).

## Todo
- [ ] update the `PaymentPointerData` to remove `received: string` and add `receivedAmount: AmountJSON`
- [ ] update the webhook open api spec
- [ ] make changes in mock ASE
- [ ] update the docs

### PR DESCRIPTION
**Pull request by Ceres445**
## Changes proposed in this pull request

- Update `PaymentPointerData` liquidity with `AmountJSON`

## Context

fixes #1650 
## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [ ] Make sure that all checks pass
- [ ] Postman collection updated
> * [x]  update the `PaymentPointerData` to remove `received: string` and add `receivedAmount: AmountJSON`
> * [x]  update the webhook open api spec
> * [ ]  make changes in mock ASE
> * [ ]  update the docs

### CODE REVIEW

**Comment by Ceres445** [comment]
I'm not sure what changes are required in Mock ASE and docs, could you please provide some more clarification on that? @sabineschaller

File: packages/backend/src/open_payments/payment_pointer/model.ts [hunk 1]
```diff
@@ -85,7 +86,11 @@ export class PaymentPointer
       paymentPointer: {
         id: this.id,
         createdAt: new Date(+this.createdAt).toISOString(),
-        received: received.toString()
+        received: {
```
Comment (sabineschaller) [inline packages/backend/src/open_payments/payment_pointer/model.ts:89]:
```suggestion
        receivedAmount: {
```

File: packages/backend/src/open_payments/payment_pointer/model.ts [hunk 2]
```diff
@@ -114,7 +119,7 @@ export type PaymentPointerData = {
   paymentPointer: {
     id: string
     createdAt: string
-    received: string
+    received: AmountJSON
```
Comment (sabineschaller) [inline packages/backend/src/open_payments/payment_pointer/model.ts:122]:
```suggestion
    receivedAmount: AmountJSON
```

File: packages/backend/src/openapi/webhooks.yaml [hunk 3]
```diff
@@ -287,8 +287,7 @@ components:
                   type: string
                   format: date-time
                 received:
-                  type: string
-                  format: uint64
+                  $ref: ./schemas.yaml#/components/schemas/amount
```
Comment (sabineschaller) [inline packages/backend/src/openapi/webhooks.yaml:290]: Call that property `receivedAmount`, please 🙂

**Review by sabineschaller — COMMENTED** [review]
Thank you for your contribution!

**Comment by sabineschaller** [comment]
> I'm not sure what changes are required in Mock ASE and docs, could you please provide some more clarification on that? @sabineschaller

Mock ASE: Looks like it doesn't handle that event at all. So no update required. I'll create an issue for that. 
Docs: please also call it `receivedAmount` [here](https://github.com/interledger/rafiki/blob/main/packages/documentation/src/content/docs/integration/event-handlers.md#payment_pointerweb_monetization)

That should be it, thank you!

**Comment by devcer** [comment]
Thank you for Contributing to Interledger during Hacktoberfest. If you have any accepted PRs, You can claim your Swag [here](https://community.interledger.org/devcer/hacktoberfest-swag-redemption-claim-your-cool-gear-2152). Last date to fill up the form is December 10th, 2023. @Ceres445

**Pull request merged**